### PR TITLE
19730: Adds CloneEntity to Amalgam API, MINOR

### PR DESCRIPTION
**Pull request by howsohazard**

### CODE REVIEW

File: src/Amalgam/entity/EntityExternalInterface.cpp [hunk 1]
```diff
@@ -80,6 +80,36 @@ EntityExternalInterface::LoadEntityStatus EntityExternalInterface::VerifyEntity(
 	return EntityExternalInterface::LoadEntityStatus(false, "", version);
 }
 
+bool EntityExternalInterface::CloneEntity(std::string &handle, std::string &cloned_handle, std::string &path, bool persistent,
```
Comment (calebwherry) [inline src/Amalgam/entity/EntityExternalInterface.cpp:83]: `persistent` isn't used.

File: src/Amalgam/entity/EntityExternalInterface.cpp [hunk 2]
```diff
@@ -80,6 +80,36 @@ EntityExternalInterface::LoadEntityStatus EntityExternalInterface::VerifyEntity(
...
+	if(bundle->entity == nullptr)
+		return false;
+
+	Entity *entity = new Entity(bundle->entity);
+	asset_manager.SetRootPermission(entity, true);
+
+	PrintListener *pl = nullptr;
+	std::vector<EntityWriteListener *> wl;
+
+	if(print_log_filename != "")
```
Comment (calebwherry) [inline src/Amalgam/entity/EntityExternalInterface.cpp]: nit: I would use !empty() but fine leaving as is.

File: src/Amalgam/entity/EntityExternalInterface.cpp [hunk 3]
```diff
@@ -80,6 +80,36 @@ EntityExternalInterface::LoadEntityStatus EntityExternalInterface::VerifyEntity(
...
+		pl = new PrintListener(print_log_filename);
+
+	if(write_log_filename != "")
```
Comment (calebwherry) [inline src/Amalgam/entity/EntityExternalInterface.cpp]: nit: I would use !empty() but fine leaving as is.

File: src/Amalgam/AssetManager.h [hunk 4]
```diff
@@ -107,6 +107,18 @@ class AssetManager
 	inline bool IsEntityDirectlyPersistent(Entity *entity)
 	{	return persistentEntities.find(entity) != end(persistentEntities);	}
 
+	//sets the entity's persistent path
+	inline void SetEntityPersistentPath(Entity *entity, std::string &resource_path)
+	{
+	#ifdef MULTITHREAD_INTERFACE
+		Concurrency::WriteLock lock(persistentEntitiesMutex);
+	#endif
+		if(resource_path == "")
```
Comment (cademack) [inline src/Amalgam/AssetManager.h]: I see you switched to use `.empty()` in a few other spots with this PR. Might wanna do that here as well if that's the way we are leaning. No big deal as far as I know though

**Review by cademack — DISMISSED** [review]

**Review by cademack — APPROVED** [review]

**Review by howsoRes — APPROVED** [review]

**Pull request merged**